core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -214,11 +214,15 @@ public void write(String s, byte[] bytes) {
 
         MethodCreator beforeAn = file.getMethodCreator("beforeAnalysis", "V", BEFORE_ANALYSIS_ACCESS);
         TryBlock overallCatch = beforeAn.tryBlock();
-        //TODO: at some point we are going to need to break this up, as if it get too big it will hit the method size limit
 
         ResultHandle beforeAnalysisParam = beforeAn.getMethodParam(0);
+
+        MethodCreator registerAsUnsafeAccessed = file
+                .getMethodCreator("registerAsUnsafeAccessed", void.class, Feature.BeforeAnalysisAccess.class)
+                .setModifiers(Modifier.PRIVATE | Modifier.STATIC);
+        overallCatch.invokeStaticMethod(registerAsUnsafeAccessed.getMethodDescriptor(), beforeAnalysisParam);
```

Comment:
It's a very minor comment but any chance you could follow the same pattern for this one and call the method after the `returnVoid`?
At first I thought something was missing.

It's really minor so feel free to ignore if you think it's not worth it.

Reply:
Done.